

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Using FishTaco via the command line &mdash; FiShTaCo 1.0.0 documentation</title>
    
    <link rel="stylesheet" href="_static/agogo.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/style.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="FiShTaCo 1.0.0 documentation" href="index.html" />
    <link rel="next" title="FishTaco Visualization" href="visualization.html" />
    <link rel="prev" title="Installation Instructions" href="installation.html" /> 
  </head>
  <body role="document">
    <div class="header-wrapper" role="banner">
      <div class="header">
        <div class="headertitle"><a
          href="index.html">FiShTaCo 1.0.0 documentation</a></div>
        <div class="rel" role="navigation" aria-label="related navigation">
          <a href="installation.html" title="Installation Instructions"
             accesskey="P">previous</a> |
          <a href="visualization.html" title="FishTaco Visualization"
             accesskey="N">next</a> |
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a>
        </div>
       </div>
    </div>

    <div class="content-wrapper">
      <div class="content">
        <div class="document">
            
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="using-fishtaco-via-the-command-line">
<h1>Using FishTaco via the command line<a class="headerlink" href="#using-fishtaco-via-the-command-line" title="Permalink to this headline">¶</a></h1>
<p id="index-0">The FishTaco python module handles all calculations internally.
FishTaco offers an interface to the FishTaco functionality via the command line and the run_fishtaco.py script.</p>
<div class="section" id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this headline">¶</a></h2>
<p><code class="docutils literal"><span class="pre">run_fishtaco.py</span> <span class="pre">-ta</span> <span class="pre">TAXA_ABUN_FILE</span> <span class="pre">-fu</span> <span class="pre">FUNCTION_ABUN_FILE</span> <span class="pre">-l</span> <span class="pre">LABELS_FILE</span> <span class="pre">[options]</span></code></p>
</div>
<div class="section" id="required-arguments">
<h2>Required arguments<a class="headerlink" href="#required-arguments" title="Permalink to this headline">¶</a></h2>
<dl class="docutils">
<dt><code class="docutils literal"><span class="pre">-ta,</span> <span class="pre">--taxa_abundance</span> <span class="pre">TAXA_ABUN_FILE</span></code></dt>
<dd>Input file of taxonomic abundance profiles (<a class="reference external" href="fishtaco_file_formats.html#taxa-abundance-file">format</a>)</dd>
<dt><code class="docutils literal"><span class="pre">-fu,</span> <span class="pre">--function_abundance</span> <span class="pre">FUNCTION_ABUN_FILE</span></code></dt>
<dd>Input file of function abundance (<a class="reference external" href="fishtaco_file_formats.html#function-abundance-file">format</a>)</dd>
<dt><code class="docutils literal"><span class="pre">-l,</span> <span class="pre">--labels</span> <span class="pre">LABELS_FILE</span></code></dt>
<dd>Input file of label assignment for the two sample sets being compared (<a class="reference external" href="fishtaco_file_formats.html#sample-sets-labels-file">format</a>)</dd>
</dl>
</div>
<div class="section" id="recommended-arguments">
<h2>Recommended arguments<a class="headerlink" href="#recommended-arguments" title="Permalink to this headline">¶</a></h2>
<dl class="docutils">
<dt><code class="docutils literal"><span class="pre">-gc,</span> <span class="pre">--genomic_content_of_taxa</span> <span class="pre">GENOMIC_CONTENT_FILE</span></code></dt>
<dd>Input file of genomic content of each taxa (<a class="reference external" href="fishtaco_file_formats.html#genomic-content-file">format</a>)</dd>
<dt><code class="docutils literal"><span class="pre">-control_label</span> <span class="pre">LABEL</span></code></dt>
<dd>Define control label (default: 0)</dd>
<dt><code class="docutils literal"><span class="pre">-case_label</span> <span class="pre">LABEL</span></code></dt>
<dd>Define case label (default: 1)</dd>
<dt><code class="docutils literal"><span class="pre">-op,</span> <span class="pre">--output_prefix</span> <span class="pre">OUTPUT_PREF</span></code></dt>
<dd>Output prefix for result files (default: fishtaco_out)</dd>
<dt><code class="docutils literal"><span class="pre">-map_function_level</span> <span class="pre">[pathway,module,none,custom]</span></code></dt>
<dd>Map KOs to pathways, modules, none, or custom (default: pathway)</dd>
</dl>
</div>
<div class="section" id="advanced-usage-arguments">
<h2>Advanced usage arguments<a class="headerlink" href="#advanced-usage-arguments" title="Permalink to this headline">¶</a></h2>
<dl class="docutils">
<dt><code class="docutils literal"><span class="pre">-map_function_file</span> <span class="pre">FUNC_LEVEL_MAP_FILE</span></code></dt>
<dd>Mapping file from KOs to pathways, modules, or custom (default: use KEGG database downloaded 07/15/2013)</dd>
<dt><code class="docutils literal"><span class="pre">-mult,</span> <span class="pre">--multiple_hypothesis_correction</span> <span class="pre">[Bonf,FDR-0.01,FDR-0.05,FDR-0.1,none]</span></code></dt>
<dd>Multiple hypothesis correction for functional enrichment (default: FDR-0.05)</dd>
<dt><code class="docutils literal"><span class="pre">-max_func,</span> <span class="pre">--maximum_functions_to_analyze</span> <span class="pre">MAX_FUNCTIONS</span></code></dt>
<dd>Maximum number of enriched functions to consider (default: All)</dd>
<dt><code class="docutils literal"><span class="pre">-assessment,</span> <span class="pre">--taxa_assessment_method</span> <span class="pre">[permute_all_but_i,permuted_shapley_orderings]</span></code></dt>
<dd>The method used when assessing taxa to compute score (default: permuted_shapley_orderings)</dd>
<dt><code class="docutils literal"><span class="pre">-score,</span> <span class="pre">--score_to_compute</span> <span class="pre">[t_test,mean_diff,median_diff,wilcoxon,log_mean_ratio]</span></code></dt>
<dd>The enrichment score to compute for each function (default: wilcoxon)</dd>
<dt><code class="docutils literal"><span class="pre">-max_score,</span> <span class="pre">--max_score_cutoff</span> <span class="pre">MAX_SCORE_CUTOFF</span></code></dt>
<dd>The maximum score cutoff (for example, when dividing by zero) (default: 100)</dd>
<dt><code class="docutils literal"><span class="pre">-na_rep</span> <span class="pre">NA_REP</span></code></dt>
<dd>How to represent NAs in the output (default: NA)</dd>
<dt><code class="docutils literal"><span class="pre">-number_of_permutations</span> <span class="pre">NUMBER_OF_PERMUTATIONS</span></code></dt>
<dd>number of permutations (default: 100)</dd>
<dt><code class="docutils literal"><span class="pre">-number_of_shapley_orderings_per_taxa</span> <span class="pre">NUMBER_OF_SHAPLEY_ORDERINGS_PER_TAXA</span></code></dt>
<dd>number of shapley orderings per taxa (default: 5)</dd>
<dt><code class="docutils literal"><span class="pre">-use_gc_as_prior,</span> <span class="pre">--use_genomic_content_of_taxa_as_prior</span></code></dt>
<dd>Learn the taxa copy number of each function, using the given genomic content data as prior (default: False)</dd>
<dt><code class="docutils literal"><span class="pre">-residual_mode</span> <span class="pre">[as_taxa,remove_residual,as_baseline]</span></code></dt>
<dd>How to treat the residual of the functional abundance profile (default: remove_residual)</dd>
<dt><code class="docutils literal"><span class="pre">-normalization_mode</span> <span class="pre">[none,scale_non_permuted,scale_permuted]</span></code></dt>
<dd>How to normalize the sample after permuting taxa (default: scale_permuted)</dd>
<dt><code class="docutils literal"><span class="pre">-permutation_mode</span> <span class="pre">[independent,blocks]</span></code></dt>
<dd>How to permute the taxa across samples (default: blocks)</dd>
<dt><code class="docutils literal"><span class="pre">-en,</span> <span class="pre">--enrichment_results</span> <span class="pre">DA_RESULT_FILE</span></code></dt>
<dd>Pre-computed functional enrichment results from the compute_differential_abundance.py script (default: None)</dd>
<dt><code class="docutils literal"><span class="pre">-single_function_filter</span> <span class="pre">SINGLE_FUNCTION_FILTER</span></code></dt>
<dd>Limit analysis only to this single function (default: None)</dd>
<dt><code class="docutils literal"><span class="pre">-h,</span> <span class="pre">--help</span></code></dt>
<dd>show help message and exit</dd>
<dt><code class="docutils literal"><span class="pre">-log,</span> <span class="pre">--log</span></code></dt>
<dd>Write to log file (default: False)</dd>
</dl>
</div>
<div class="section" id="fishtaco-output-files">
<h2>FishTaco Output Files<a class="headerlink" href="#fishtaco-output-files" title="Permalink to this headline">¶</a></h2>
<div class="section" id="main-output-files">
<h3>Main output files<a class="headerlink" href="#main-output-files" title="Permalink to this headline">¶</a></h3>
<dl class="docutils">
<dt><code class="docutils literal"><span class="pre">fishtaco_out_main_output_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the taxon-level decomposition of shift scores for the differentially abundant functions. (<a class="reference external" href="fishtaco_file_formats.html#fishtaco-output-file">format</a>)</dd>
</dl>
</div>
<div class="section" id="supporting-stats-output-files">
<h3>Supporting stats output files<a class="headerlink" href="#supporting-stats-output-files" title="Permalink to this headline">¶</a></h3>
<dl class="docutils">
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_taxa_contributions_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the final taxon-level contribution score for every differentially abundant(shifted) function in the input data, as calculated by FishTaco</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_DA_function_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains statistics regarding the differential abundance for each function in the input file</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_DA_taxa_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains statistics regarding the differential abundance for each taxa in the input file</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_mean_stat_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the mean taxon-level contribution score for every differentially abundant(shifted) function in the input data (in default settings, this is equal to the final score)</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_median_stat_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the median taxon-level contribution score for every differentially abundant(shifted) function in the input data</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_std_stat_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the standard deviation of taxon-level contribution score for every differentially abundant(shifted) function in the input data</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_original_value_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the metagenome-based shift statistics value for each function in the input file</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_predicted_DA_value_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the taxa-based shift statistics value for each function in the input file</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_predicted_function_abundance_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the taxa-based abundance profile for each function in each sample</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_predicted_function_agreement_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains various statistics regarding the agreement between the metagenome- and taxa-based abundance profiles for each function</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_residual_function_abundance_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the residual between the metagenome- and taxa-based abundance profiles for each function (in &#8220;remove-residual&#8221; mode the residual is equal to zero)</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_shapley_orderings_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the random Shapley orderings used in the run (for &#8220;permuted_shapley_orderings&#8221; mode)</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_taxa_learned_copy_num_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the inferred copy numbers of each function in each taxon (for FishTaco with prior-based or <em>de novo</em> inference)</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_taxa_learning_rsqr_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains various statistics regarding the agreement between the metagenome- and taxa-based abundance profiles for each function (on test data)</dd>
<dt><code class="docutils literal"><span class="pre">fishtaco_out_STAT_run_log_SCORE_wilcoxon_ASSESSMENT_permuted_shapley_orderings.tab</span></code></dt>
<dd>contains the running log of FishTaco</dd>
</dl>
</div>
</div>
<div class="section" id="examples">
<h2>Examples<a class="headerlink" href="#examples" title="Permalink to this headline">¶</a></h2>
<p>The <em>fishtaco/examples</em> directory contains the following files:</p>
<ul class="simple">
<li>the file <em>METAPHLAN_taxa_vs_SAMPLE_for_K00001.tab</em> contains scaled abundance measurements of 10 species in 213 samples from the HMP dataset</li>
<li>the file <em>WGS_KO_vs_SAMPLE_MUSiCC_only_K00001.tab</em> contains MUSiCC-corrected abundance values for the K00001 orthology group in the same samples</li>
<li>the file <em>METAPHLAN_taxa_vs_KO_only_K00001.tab</em> contains the copy numbers of the K00001 orthology group in the 10 species as above</li>
<li>the file <em>SAMPLE_vs_CLASS.tab</em> contains class labels from the same samples (control vs. case)</li>
</ul>
<p>Using these files as input for FishTaco results in the following output files (found in the <em>fishtaco/examples/output</em> directory):</p>
<div class="section" id="fishtaco-with-no-inference-output-fishtaco-out-no-inf-stat">
<h3>FishTaco with no inference (output/fishtaco_out_no_inf_STAT_*)<a class="headerlink" href="#fishtaco-with-no-inference-output-fishtaco-out-no-inf-stat" title="Permalink to this headline">¶</a></h3>
<div class="code python highlight-python"><div class="highlight"><pre>run_fishtaco.py -ta fishtaco/examples/METAPHLAN_taxa_vs_SAMPLE_for_K00001.tab
-fu fishtaco/examples/WGS_KO_vs_SAMPLE_MUSiCC_only_K00001.tab
-t2f fishtaco/examples/METAPHLAN_taxa_vs_KO_only_K00001.tab
-c fishtaco/examples/SAMPLE_vs_CLASS.tab -op fishtaco_out_no_inf -max_da 1
-assessment permuted_shapley_orderings -score wilcoxon -na_rep 0
-number_of_shapley_orderings_per_taxa 3 -residual_mode remove_residual -log
-normalization_mode scale_permuted -permutation_mode blocks -number_of_permutations 5
</pre></div>
</div>
</div>
<div class="section" id="fishtaco-with-prior-based-inference-output-fishtaco-out-prior-based-inf-stat">
<h3>FishTaco with prior-based inference (output/fishtaco_out_prior_based_inf_STAT_*)<a class="headerlink" href="#fishtaco-with-prior-based-inference-output-fishtaco-out-prior-based-inf-stat" title="Permalink to this headline">¶</a></h3>
<div class="code python highlight-python"><div class="highlight"><pre>run_fishtaco.py -op fishtaco_out_no_inf -max_da 1
-ta fishtaco/examples/METAPHLAN_taxa_vs_SAMPLE_for_K00001.tab
-fu fishtaco/examples/WGS_KO_vs_SAMPLE_MUSiCC_only_K00001.tab
-c fishtaco/examples/SAMPLE_vs_CLASS.tab
-t2f fishtaco/examples/METAPHLAN_taxa_vs_KO_only_K00001.tab
-assessment permuted_shapley_orderings -score wilcoxon
-na_rep 0 -number_of_shapley_orderings_per_taxa 3 -residual_mode remove_residual
-normalization_mode scale_permuted -permutation_mode blocks -number_of_permutations 5
-use_t2f_as_prior -log
</pre></div>
</div>
</div>
<div class="section" id="fishtaco-with-de-novo-inference-output-fishtaco-out-de-novo-inf-stat">
<h3>FishTaco with de novo inference (output/fishtaco_out_de_novo_inf_STAT_*)<a class="headerlink" href="#fishtaco-with-de-novo-inference-output-fishtaco-out-de-novo-inf-stat" title="Permalink to this headline">¶</a></h3>
<div class="code python highlight-python"><div class="highlight"><pre>run_fishtaco.py -op fishtaco_out_no_inf -max_da 1
-ta fishtaco/examples/METAPHLAN_taxa_vs_SAMPLE_for_K00001.tab
-fu fishtaco/examples/WGS_KO_vs_SAMPLE_MUSiCC_only_K00001.tab
-c fishtaco/examples/SAMPLE_vs_CLASS.tab -assessment permuted_shapley_orderings
-score wilcoxon -na_rep 0 -number_of_shapley_orderings_per_taxa 3
-residual_mode remove_residual -normalization_mode scale_permuted
-permutation_mode blocks -number_of_permutations 5 -log
</pre></div>
</div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
        </div>
        <div class="sidebar">
          <h3>Table Of Contents</h3>
          <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation Instructions</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Using FishTaco via the command line</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualization.html">FishTaco Visualization</a></li>
</ul>

          <div role="search">
            <h3 style="margin-top: 1.5em;">Search</h3>
            <form class="search" action="search.html" method="get">
                <input type="text" name="q" />
                <input type="submit" value="Go" />
                <input type="hidden" name="check_keywords" value="yes" />
                <input type="hidden" name="area" value="default" />
            </form>
            <p class="searchtip" style="font-size: 90%">
                Enter search terms or a module, class or function name.
            </p>
          </div>
        </div>
        <div class="clearer"></div>
      </div>
    </div>

    <div class="footer-wrapper">
      <div class="footer">
        <div class="left">
          <div role="navigation" aria-label="related navigaton">
            <a href="installation.html" title="Installation Instructions"
              >previous</a> |
            <a href="visualization.html" title="FishTaco Visualization"
              >next</a> |
            <a href="genindex.html" title="General Index"
              >index</a>
          </div>
          <div role="note" aria-label="source link">
              <br/>
              <a href="_sources/execution.txt"
                rel="nofollow">Show Source</a>
          </div>
        </div>

        <div class="right">
          
    <div class="footer" role="contentinfo">
        &copy; Copyright 2015, Ohad Manor.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3b2.
    </div>
        </div>
        <div class="clearer"></div>
      </div>
    </div>

  </body>
</html>